elastic/kibana
# [Index lifecycle management] Add deprecated policy warnings

Partially addresses https://github.com/elastic/kibana/issues/170805

## Summary

With https://github.com/elastic/elasticsearch/pull/101148 we now can alert the users when an ILM policy will soon be deprecated and shouldnt be relied upon. This PR adds a badge and a callout to the ILM UI to alert the users about this.

##### How to test

* Start up kibana and es
* Create a deprecated policy using devtools
* Navigate to Stack management -> ILM
* Verify that:
  * Deprecated policies are hidden by default
  * Upon enabling filtering to see them they should pop up on the table with a badge
  * Verify that deprecated policies are highlighted and when editing, we also show a warning callout.

<details>
<summary>Create deprecated policy</summary>

```
PUT _ilm/policy/my_policy
{
  "policy": {
      "deprecated": true,
    "_meta": {
      "description": "used for nginx log",
      "project": {
        "name": "myProject",
        "department": "myDepartment"
      }
    },
    "phases": {
      "warm": {
        "min_age": "10d",
        "actions": {
          "forcemerge": {
            "max_num_segments": 1
          }
        }
      },
      "delete": {
        "min_age": "30d",
        "actions": {
          "delete": {}
        }
      }
    }
  }
}
```

</details>


<details>
<summary>Screenshots</summary>

<img width="1508" alt="Screenshot 2024-01-03 at 11 09 58" src="https://github.com/elastic/kibana/assets/1191206/f9eb7d67-1bb9-40ed-b6b3-d39e96fb9b71">

![Screenshot 2024-01-10 at 11 34 00](https://github.com/elastic/kibana/assets/1191206/dc09a52b-3615-4479-9573-701a5e8153a3)
![Screenshot 2024-01-10 at 11 34 07](https://github.com/elastic/kibana/assets/1191206/7c2c4722-7fa7-4d67-8c42-cba1c5b96488)




</details>
